processing/src/main/java/org/apache/druid/query/operator/GlueingPartitioningOperator.java

```diff
@@ -0,0 +1,279 @@
+    private int currentIndex = 0;
+    private boolean firstPartitionHandled = false;
+
+    public GluedRACsIterator(RowsAndColumns rac)
+    {
+      this.rac = rac;
+      ClusteredGroupPartitioner groupPartitioner = rac.as(ClusteredGroupPartitioner.class);
+      if (groupPartitioner == null) {
+        groupPartitioner = new DefaultClusteredGroupPartitioner(rac);
+      }
+      this.boundaries = groupPartitioner.computeBoundaries(partitionColumns);
+    }
+
+    @Override
+    public boolean hasNext()
+    {
+      return currentIndex < boundaries.length - 1;
+    }
+
+    /**
+     * Retrieves the next partition in the RowsAndColumns. If the first partition has not been handled yet,
+     * it may be glued with the previous RowsAndColumns if the partition columns match.
+     *
+     * @return The next RowsAndColumns partition, potentially glued with the previous one.
+     * @throws NoSuchElementException if there are no more partitions.
+     */
+    @Override
+    public RowsAndColumns next()
+    {
+      if (!hasNext()) {
+        throw new NoSuchElementException();
+      }
+
+      if (!firstPartitionHandled) {
+        firstPartitionHandled = true;
+        int start = boundaries[currentIndex];
+        int end = boundaries[currentIndex + 1];
+        LimitedRowsAndColumns limitedRAC = new LimitedRowsAndColumns(rac, start, end);
+
+        if (isGlueingNeeded(previousRac, limitedRAC)) {
+          RowsAndColumns gluedRAC = getConcatRacForFirstPartition(previousRac, limitedRAC);
+          ensureMaxRowsMaterializedConstraint(gluedRAC.numRows());
+          previousRac = null;
+          currentIndex++;
+          return gluedRAC;
+        } else {
+          if (previousRac != null) {
+            RowsAndColumns temp = previousRac;
+            previousRac = null;
+            return temp;
+          }
+        }
+      }
+
+      int start = boundaries[currentIndex];
+      int end = boundaries[currentIndex + 1];
+      currentIndex++;
+      return new LimitedRowsAndColumns(rac, start, end);
+    }
+
+    /**
+     * Determines whether gluing is needed between the previous RowsAndColumns and the first partition
+     * of the current RowsAndColumns based on the partition columns. If the columns match, the two RACs
+     * will be glued together.
+     *
+     * @param previousRac The previous RowsAndColumns.
+     * @param firstPartitionOfCurrentRac The first partition of the current RowsAndColumns.
+     * @return true if gluing is needed, false otherwise.
+     */
+    private boolean isGlueingNeeded(RowsAndColumns previousRac, RowsAndColumns firstPartitionOfCurrentRac)
+    {
+      if (previousRac == null) {
+        return false;
+      }
+
+      final ConcatRowsAndColumns concatRac = getConcatRacForFirstPartition(previousRac, firstPartitionOfCurrentRac);
+      for (String column : partitionColumns) {
+        final Column theCol = concatRac.findColumn(column);
+        if (theCol == null) {
+          continue;
```

Comment:
why not error or `true`?

Reply:
This was done similar to https://github.com/apache/druid/blob/88d26e4541e31dd3107f120d24f9e7c21db160a4/processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultClusteredGroupPartitioner.java#L56.

We cannot return true here since some other column might mismatch. I'm not sure why we don't throw error in such cases, but I kept it similar to the existing flows.

Reply:
I think if the column on which we supposed to partition doesn't exists should be an error; does this happen?

Reply:
I don't think so, have made the change to throw error.